src/ast.ts

```diff
@@ -36,20 +36,26 @@ export interface ElementNodeProto {
   children: Array<NodeProto>;
 }
 
+export interface DocumentNodeProto {
+  tagid: 92; // See htmltagenum.ts
+  children: Array<NodeProto>;
+}
+
 export type NodeProto = TextNodeProto | ElementNodeProto;
 export interface AttributeProto {
   name: string;
   value?: string;
 }
 
 export function isElementNode(node: NodeProto): node is ElementNodeProto {
-  return 'tagid' in node;
+  return (node as any).tagid !== undefined;
 }
 
 export function fromDocument(doc: Document): TreeProto {
+  const children = Array.from(doc.childNodes).map(mapDomNodeToNodeProto);
   return {
     quirks_mode: doc.compatMode === 'BackCompat',
-    tree: Array.from(doc.childNodes).map(mapDomNodeToNodeProto),
+    tree: [getDocumentNode(children)],
```

Comment:
tbh I think inlining this would be easier. The extra function/indirection doesn't abstract away anything particularly useful imo, and the typechecker will ensure that the doc node is always included. If anything, it makes sense to make it a test-helper (since that's where it's really used), but include it directly here.